solr/core/src/java/org/apache/solr/spelling/suggest/fst/AnalyzingInfixLookupFactory.java

```diff
@@ -89,7 +88,7 @@ public Lookup create(NamedList<?> params, SolrCore core) {
     String indexPath =
         params.get(INDEX_PATH) != null ? params.get(INDEX_PATH).toString() : DEFAULT_INDEX_PATH;
     if (!Path.of(indexPath).isAbsolute()) {
-      indexPath = core.getDataDir() + File.separator + indexPath;
+      indexPath = Path.of(core.getDataDir(), indexPath).toString();
```

Comment:
can indexPath be a Path?   Also, is there an opportunity to align variable naming between indexPath and the use of indexDir in in AbstractLuceneSpellChecker?